Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -124,8 +123,20 @@ public BlackboardArtifactNode(BlackboardArtifact artifact, String iconPath) {
         super(Children.LEAF, createLookup(artifact));
 
         this.artifact = artifact;
-        //this.associated = getAssociatedContent(artifact);
-        this.associated = this.getLookup().lookup(Content.class);
+        
+        // Look for associated Content  i.e. the source file for the artifact
+        if (this.getLookup().lookupAll(Content.class).size() > 1) {
+            for (Content content : this.getLookup().lookupAll(Content.class)) {
+                if ( (content != null)  && (!(content instanceof BlackboardArtifact)) ){
+                    this.associated = content;
+                    break;
+                }
+            }
+        }
+        if (null == this.associated ) {
+            this.associated = this.getLookup().lookup(Content.class);
+        }
+        
```

Comment:
I'm fixing this as part of "Show messages in Data source tree" work,  Your point about not needing to call lookupAll() twice is well taken. The way I had it, we were looping through the contents returned by lookupAll only if we knew there was more than 1.  Hence the second call to lookup.

The two constructors for BlackboardArtifactNode are nearly identical, one should simply call the other passing the assumed iconpath, that will eliminate the duplication.